domains/pallets/messenger/src/messages.rs

```diff
@@ -125,6 +126,10 @@ impl<T: Config> Pallet<T> {
                 let response = if let Some(endpoint_handler) =
                     T::get_endpoint_handler(&req.dst_endpoint)
                 {
+                    if !ChainAllowlist::<T>::get().contains(&dst_chain_id) {
+                        return Err(Error::<T>::ChainNotAllowed.into());
+                    }
```

Comment:
I'm confused, this seems like returning the error as the result of extrinsic execution, not returning an error response to the src chain.

Reply:
looks like this was a regression before this PR if you follow the code.
Should be updated to ensure EndpointResponse is returned instead of failing the dispatch